symengine/mul.cpp

```diff
@@ -363,8 +363,8 @@ RCP<const Basic> mul_expand_two(const RCP<const Basic> &a, const RCP<const Basic
             rcp_static_cast<const Add>(b)->coef_);
         umap_basic_num d;
         // Improves (x+1)**3*(x+2)**3*...(x+350)**3 expansion from 0.97s to 0.93s:
-        d.reserve((rcp_static_cast<const Add>(a))->dict_.size()*
-            (rcp_static_cast<const Add>(b))->dict_.size());
+//        d.reserve((rcp_static_cast<const Add>(a))->dict_.size()*
+//            (rcp_static_cast<const Add>(b))->dict_.size());
```

Comment:
For some reason, `std::unordered_map` doesn't have the `reserve()` method on Intel. I need to see what is going on. One workaround could be to write a function `unordered_map_reserve` and in it just use some ifdefs. In the long run, we probably want to use our own hashtable anyway, so this issue will disappear.